core/src/main/scala/quasar/plugin/postgres/package.scala

```diff
@@ -78,6 +89,57 @@ package object postgres {
   def hygienicIdent(ident: Ident): Ident =
     s""""${ident.replace("\"", "\"\"")}""""
 
+  def specifyColumnFragments[F[_]: ApplicativeError[?[_], Throwable]](
+    cols: NonEmptyList[Column[ColumnType.Scalar]])
+      : F[NonEmptyList[Fragment]] =
+    cols.traverse(columnSpec).fold(
+      invalid => ApplicativeError[F, Throwable].raiseError(new ColumnTypesNotSupported(invalid)),
+      _.pure[F])
+
+  def specifyColumnFragment[F[_]: ApplicativeError[?[_], Throwable]](
+    col: Column[ColumnType.Scalar])
+      : F[Fragment] =
+    columnSpec(col).fold(
+      invalid => ApplicativeError[F, Throwable].raiseError(new ColumnTypesNotSupported(invalid)),
+      _.pure[F])
+
+  def createTable(log: Logger)(table: Table, colSpecs: NonEmptyList[Fragment])
+      : ConnectionIO[Int] = {
+    val preamble =
+      fr"CREATE TABLE IF NOT EXISTS" ++ Fragment.const(hygienicIdent(table))
+
+    (preamble ++ Fragments.parentheses(colSpecs.intercalate(fr",")))
+      .updateWithLogHandler(logHandler(log))
+      .run
+  }
+
+  def createTableIfNotExists(log: Logger)(table: Table, colSpecs: NonEmptyList[Fragment]): ConnectionIO[Int] = {
+    val preamble =
+      fr"CREATE TABLE IF NOT EXISTS" ++ Fragment.const(hygienicIdent(table))
+
+    (preamble ++ Fragments.parentheses(colSpecs.intercalate(fr",")))
+      .updateWithLogHandler(logHandler(log))
+      .run
+  }
+
+  def createIndex(log: Logger)(table: Table, col: Fragment): ConnectionIO[Int] =
+    ((fr"CREATE INDEX IF NOT EXISTS __precog_ix__ ON" ++
```

Comment:
Postgres' `create index if not exists` is not very smart. It asks for a name and avoids creating an index with the same name again. It doesn't really check if the index structure matches the one you're attempting to create. Hopefully this name will be unique so we don't create it more than once.